File: pykanum.java
public class pykanum{

    private char[] num;
    private int size;

    public pykanum(){
        nonRepRandom n = new nonRepRandom(9);
        size = 4;
        num = new char[size];
        for(int i = 0;i <= size -1; i++){
            num[i] = (char)(n.generateNum() + '0');
        }
    }

    public pykanum(int s){
        nonRepRandom n = new nonRepRandom(9);
        size = s;
        num = new char[size];
        for(int i = 0;i <= size -1; i++){
            num[i] = (char)(n.generateNum() + '0');
        }
    }

    public pykanum(String n){
        size = n.length();
        num = new char[size];
        for(int i = 0; i < size; i++){
            num[i] = n.charAt(i);
        }
        
    }
    private boolean inArray(int n){
        for(int i = 0; i <= size - 1; i++){
            if(n == num[i]){
                return true;
            }
        }
        return false;
    }

    public String compareNums(String guess){
        int pyka = 0;
        int centra = 0;
        pykanum g = new pykanum(guess);
        for(int i = 0; i <= size - 1; i++){
            if(g.num[i] == num[i]){
                centra++;
            } else if(inArray(g.num[i])){
                pyka++;
            }
        }
        return pyka + " pyka, " + centra + " centra.";
    }

    public String toString(){
        String output = "";
        for(int i = 0; i <= size -1 ; i++){
            output += ("" + num[i]);
        }
        return output;
    }
    
}

File: python-code/Pykanum.py
import NonRepRandom


class PykaNum:
    def __init__(self, size=4, pnum=None):
        if pnum == None:  # genreate a new pykanum if there isn't one given
            self.size = size
            num = []
            n = NonRepRandom()
            for i in range(size):
                num.append(n.generateNum())
        else: # if a num is given, turn that num into a pyka num
            for x in pnum:
                self.size = pnum.len
                num.insert(0, x)

    def CompareNums(self, other):
        pyka = 0
        centra = 0
        guess = PykaNum(other)
        for i in self.size :
            if self.num[i] == other.num[i] :
                centra += 1
            elif other.num.contains(self.num[i]):
                pyka += 1
        return f"{pyka} pyka and {centra} centra."